heat/core/communication.py

```diff
@@ -463,6 +464,161 @@ def Iallgatherv(self, sendbuf, recvbuf, send_axis=0, recv_axis=None):
         return self.__allgather_like(self.handle.Iallgatherv, sendbuf, recvbuf, send_axis, recv_axis)
     Iallgatherv.__doc__ = MPI.Comm.Iallgatherv.__doc__
 
+    @classmethod
+    def alltoall_sendbuffer(cls, obj, nproc, counts=None, displs=None): #This is a temporary soltion to pass number of arguments as variable. Better to access via cls itself, but how?
+        mpi_type, elements = cls.__mpi_type_mappings[obj.dtype], torch.numel(obj)
+
+        shape = obj.shape
+        strides = [1] * len(shape)
+        strides[-1] = obj.stride()[-1]
+
+        offsets = [0]*len(shape)
+        offsets[1:] = [obj.element_size() * stride for stride in obj.stride()[:-1]]
+
+        # Step 1: Wrap along axes > 1 (all axes except send_axis and recv_axis
+        for i in range(len(shape) - 1, 1, -1):
+            mpi_type = mpi_type.Create_vector(shape[i], 1, strides[i]).Create_resized(0, offsets[i])
+            mpi_type.Commit()
+
+        # Step 2: Create Custom sized vector datatypes, according to rank-specific size along send_axis
+        # send_elements has nproc entries, defining how many vectors of mpi_type are stacked together for each process to receive along the send_axis
+        send_elements = np.full((nproc,), obj.shape[1]//nproc)
+        send_elements[:obj.shape[1]%nproc]+=1
+
+        #Create short_Type from the last entry of send_elements
+        mpi_short_type = mpi_type.Create_vector(send_elements[-1], 1, strides[1]).Create_resized(0, offsets[1])
+        mpi_short_type.Commit()
+        #Create long_Type from the first entry of send_elements (wraps one more mpi_type vector than short_Type
+        mpi_long_type = mpi_type.Create_vector(send_elements[0], 1, strides[1]).Create_resized(0, offsets[1])
+        mpi_long_type.Commit()
+
+        #Step 3: Pack short_type and long_type along the recv_axis
+        mpi_short_type=mpi_short_type.Create_vector(shape[0], 1, strides[0]).Create_resized(0, send_elements[-1] * obj.stride()[1] * obj.element_size())
```

Comment:
I am certain that chaining type creation operations are fine in Fortran. I am not sure whether this is the case for C/C++ implementation. Since we bind against them that might be a problem for certain MPI stacks. Need to keep an eye on this. Could lead some bugs

Reply:
? But this is what is done in all the other cases as well, you always did
mpi_type = mpi_type.Create_vector
Plus, so far it works ^^

Reply:
okay so we leave it, maybe add short comment of the sorts

# XXX: might not work for all MPI stacks. Might require multiple type commits or so

This will ease the search for other future developers